src/plots/gl3d/scene.js

```diff
@@ -725,45 +726,44 @@ proto.plot = function(sceneData, fullLayout, layout) {
         });
     }
 
-    var axesScaleRatio = [1, 1, 1];
-
-    // Compute axis scale per category
-    for(i = 0; i < 3; ++i) {
-        axis = fullSceneLayout[axisProperties[i]];
-        axisType = axis.type;
-        var axisRatio = axisTypeRatios[axisType];
-        axesScaleRatio[i] = Math.pow(axisRatio.acc, 1.0 / axisRatio.count) / dataScale[i];
-    }
-
     /*
      * Dynamically set the aspect ratio depending on the users aspect settings
      */
-    var axisAutoScaleFactor = 4;
     var aspectRatio;
-
-    if(fullSceneLayout.aspectmode === 'auto') {
-        if(Math.max.apply(null, axesScaleRatio) / Math.min.apply(null, axesScaleRatio) <= axisAutoScaleFactor) {
-            /*
-             * USE DATA MODE WHEN AXIS RANGE DIMENSIONS ARE RELATIVELY EQUAL
-             */
-
-            aspectRatio = axesScaleRatio;
-        } else {
-            /*
-             * USE EQUAL MODE WHEN AXIS RANGE DIMENSIONS ARE HIGHLY UNEQUAL
-             */
-            aspectRatio = [1, 1, 1];
-        }
-    } else if(fullSceneLayout.aspectmode === 'cube') {
+    var aspectmode = scene._aspectmode || fullSceneLayout.aspectmode;
```

Comment:
What's the difference between `scene._aspectmode` and `fullSceneLayout.aspectmode` ?

Reply:
`scene._aspectmode` is an internal variable which is set after `aspectratio` values are computed.

Reply:
Are there situations where `scene._aspectmode` and `fullSceneLayout.aspectmode` aren't the same?

Reply:
They are. Namely after using scroll zoom.

Reply:
Ok, but shouldn't the changes to `aspectmode` during scroll zoom propagate to `fullLayout` ?

Reply:
I initially tried to achieve that on another branch using `relayout` updates. 
Since tests started failing, I thought that may not be ideal to override `fullLayout` in this case.

Reply:
I'm curious. Which tests fail?

Reply:
> Are there situations where `scene._aspectmode` and `fullSceneLayout.aspectmode` aren't the same?

OK. This time I was able to make that work without a temporary variable: https://github.com/plotly/plotly.js/pull/4578/commits/1828797e27f6394dc4f27a7be1fe0f6a762d5037